tests/streamlit_ui_tests/toolbar/top.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { expect, test } from '@playwright/test';
+import { clickButtonAndAwaitResponse, getMitoFrameWithTestCSV } from '../utils';
+
+test.describe('Top of Toolbar', () => {
+    test('Undo', async ({ page }) => {
+        const mito = await getMitoFrameWithTestCSV(page);
+        await expect(mito.getByText('Import Files')).not.toBeVisible();
+
+        await mito.getByTitle(/Undo the most recent edit./).click();
+        await expect(mito.getByText('Import Files')).toBeVisible();
+    });
+
+    test('Redo', async ({ page }) => {
+        const mito = await getMitoFrameWithTestCSV(page);
+        await expect(mito.getByText('Import Files')).not.toBeVisible();
+
+        await mito.getByTitle(/Undo the most recent edit./).click();
+        await expect(mito.getByText('Import Files')).toBeVisible();
+
+        await mito.getByTitle(/Reapplies the last step that you undid, as long as you haven't made any edits since the undo./).click();
+        await expect(mito.getByText('Import Files')).not.toBeVisible();
+    });
+
+    test('Clear', async ({ page }) => {
+        const mito = await getMitoFrameWithTestCSV(page);
+
+        // Insert a new column
+        await mito.locator('.mito-toolbar-button', { hasText: 'Insert' }).click();
+        await expect(mito.locator('.endo-column-header-final-text', { hasText: /new-column/ })).toBeVisible();
+
+        // Clear the analysis
+        await mito.getByTitle('Removes all of the transformations you\'ve made to imported dataframes.').click();
+        await expect(mito.getByText('Clear your current analysis?')).toBeVisible();
+        await mito.getByText('Clear', { exact: true }).click();
+
+        // Expect that the column does not exist
+        await expect(mito.locator('.endo-column-header-final-text', { hasText: /new-column/ })).not.toBeVisible();
+    });
+
+    test('Steps', async ({ page }) => {
```

Comment:
Do you want to test that we can:
1. check out a previous step by clicking on it in the taskpane
2. see the popup in the left hand corner that tells us to go back to the end of the analysis
3. clicking on the button in the left hand corner takes us back to the end of the analysis 


.. we could also do this in a separate taskpane test though